src/app/shared/components/pagination/pagination.component.ts

```diff
@@ -1,24 +1,25 @@
-import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
 import { MatPaginatorModule, PageEvent } from '@angular/material/paginator';
 
 @Component({
   selector: 'app-pagination',
   templateUrl: './pagination.component.html',
   styleUrls: ['./pagination.component.scss'],
   imports: [MatPaginatorModule],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class PaginationComponent {
   @Input() length!: number | null;
   @Output() pokemonsCountRequest = new EventEmitter<{ pageIndex: number; pageSize: number }>();
 
-  pageSize = 20;
-  pageSizeOptions = [5, 10, 20];
-  pageIndex = 0;
+  public pageSize = 20;
+  public pageSizeOptions = [5, 10, 20];
+  public pageIndex = 0;
 
-  hidePageSize = false;
-  showPageSizeOptions = true;
-  showFirstLastButtons = true;
-  disabled = false;
+  public hidePageSize = false;
+  public showPageSizeOptions = true;
+  public showFirstLastButtons = true;
+  public disabled = false;
 
   pageEvent!: PageEvent;
```

Comment:
public or not public

Reply:
public . because these properties are applied to attributes in html (to customize the appearance). although at the moment they are not involved in html

Reply:
i am about pageEvent

Reply:
49d42a8